msticpy/data/drivers/local_velociraptor_driver.py

```diff
@@ -0,0 +1,222 @@
+        self._schema: Dict[str, Any] = {}
+        self._query_map: Dict[str, List[str]]
+        self._progress = kwargs.pop("progress", True)
+        self._loaded = True
+        self.has_driver_queries = True
+        logger.info("data files to read %s", ",".join(self.data_files))
+
+    def connect(self, connection_str: Optional[str] = None, **kwargs):
+        """
+        Connect to data source.
+
+        Parameters
+        ----------
+        connection_str : str
+            Connect to a data source
+
+        """
+        del connection_str
+        self.data_files = self._get_logfile_paths()
+        self._connected = True
+
+    @property
+    def schema(self) -> Dict[str, Dict]:
+        """
+        Return current data schema of connection.
+
+        Returns
+        -------
+        Dict[str, Dict]
+            Data schema of current connection.
+
+        """
+        if not self._schema:
+            if not self.data_files:
+                self.connect()
+            # read the first row of each file to get the schema
+            iter_data_files = (
+                tqdm(self.data_files.items())
+                if self._progress
+                else self.data_files.items()
+            )
+            for table, files in iter_data_files:
+                if not files:
+                    continue
+                sample_df = pd.read_json(files[0], lines=True, nrows=1)
+                self._schema[table] = {
+                    col: dtype.name for col, dtype in sample_df.dtypes.to_dict().items()
+                }
+            logger.info("Reading schema for %d tables", len(self.data_files))
+
+        return self._schema
+
+    def query(
+        self, query: str, query_source: Optional[QuerySource] = None, **kwargs
+    ) -> Union[pd.DataFrame, Any]:
+        """
+        Execute query string and return DataFrame of results.
+
+        Parameters
+        ----------
+        query : str
+            The query to execute
+        query_source : QuerySource
+            The query definition object
+
+        Returns
+        -------
+        Union[pd.DataFrame, results.ResultSet]
+            A DataFrame (if successful) or
+            the underlying provider result if an error.
+
+        """
+        del kwargs
+        if not self.data_files:
+            self.connect()
+
+        if query not in self.data_files:
+            raise MsticpyDataQueryError(
+                f"No data loaded for query {query}.",
+                "Please check that the data loaded from the log paths",
+                "has this data type.",
+            )
+        return self._cached_query(query)
+
+    @lru_cache(maxsize=256)
+    def _cached_query(self, query: str) -> pd.DataFrame:
+        iter_data_files = (
+            tqdm(self.data_files[query]) if self._progress else self.data_files[query]
+        )
+        dfs = [pd.read_json(file, lines=True) for file in iter_data_files]
+        query_df = pd.concat(dfs)
```

Comment:
Unless there is a reason to preserve the index, I recommend passing `ignore_index=True` to `pd.concat` here. This ensures a unique index per row of concatenated results.

Reply:
Good catch